uportal-war/src/main/webapp/WEB-INF/flows/entity-selector/selectEntities.jsp

```diff
@@ -139,180 +126,280 @@
+                </div> <!-- end .modal-dialog div -->
+            </div><!--end adhocGroupModal -->
+        </div> <!-- end #content div -->
+    </div><!--end portlet-->
+
+    <script type="text/javascript">
+    </script>
+
     <script type="text/javascript">
+        var setGroupDescription = function() {
+            var includes, excludes, description;
+            descInput = $("#groupDesc");
+            includes = $("li", "#${n}dataIncludesList").map(function () { return $(this).text(); }).get();
+            excludes = $("li", "#${n}dataExcludesList").map(function () { return $(this).text(); }).get();
+
+            if (includes.length + excludes.length == 0) {
+                $("#groupDesc").val("Collection of ad hoc groups");
+                return;
+            }
+
+            description = "Users who";
+
+            if (includes.length > 0) {
+                description += " are members of ("
+                description += includes.toString();
+                description += ")";
+            }
+
+            if (excludes.length > 0) {
+                if (includes.length > 0) {
+                    description += " but";
+                }
+                description += " are not members of ("
+                description += excludes.toString();
+                //trim last ,
+                description += ")";
+            }
+
+            $("#groupDesc").val(description);
+        };
+
+        var setJSTreeEventListeners = function() {
+            $("#${n}dataIncludes").on("changed.jstree", function (e, data) {
+                var list = $("#${n}dataIncludesList");
+                list.empty();
+                $.each(data.selected, function (i, name) {
+                    var li = $("<li/>").text(data.instance.get_node(name).text).appendTo(list);
+                });
+                setGroupDescription();
+            });
+
+            $("#${n}dataExcludes").on("changed.jstree", function (e, data) {
+                var list = $("#${n}dataExcludesList");
+                list.empty();
+                $.each(data.selected, function (i, name) {
+                    var li = $("<li/>").text(data.instance.get_node(name).text).appendTo(list);
+                });
+                setGroupDescription();
+            });
+
+            $("#${n}dataIncludes").on("loaded.jstree", resetAdHocDialog);
+            $("#${n}dataExcludes").on("loaded.jstree", resetAdHocDialog);
+        };
+
+        var createTestMaps = function (groupNames, testAttr) {
+            var tests = [];
+            $.each(groupNames, function (i, name) {
+                tests.push( { "testValue": name, "attributeName": testAttr,
+                              "testerClassName" : "org.jasig.portal.groups.pags.testers.AdHocGroupTester" } );
+            });
+            return tests;
+        };
+
+        var resetAdHocDialog = function () {
+            console.log("resetting ad hoc dialog");
```

Comment:
Not sure. There was one in the original browseEntity() function. These should be stripped out.